src/map/entities/battleentity.cpp

```diff
@@ -566,6 +566,12 @@ uint16 CBattleEntity::ATT()
     if (this->objtype & TYPE_PC)
     {
         ATT += GetSkill(m_Weapons[SLOT_MAIN]->getSkillType()) + m_Weapons[SLOT_MAIN]->getILvlSkill();
+
+        // Smite applies when using 2H or H2H weapons
+        if (m_Weapons[SLOT_MAIN]->isTwoHanded() || m_Weapons[SLOT_MAIN]->getSkillType() == SKILL_H2H)
+        {
+            ATT += static_cast<int32>(ATT * this->getMod(Mod::SMITE) / 256.f); // Divide smite value by 256
```

Comment:
I really like that we treat ATT as int32 and return an uint16.

Reply:
Thats just what the phone company wants you to think before the bill arrives!

..darn it dsp and your crappy function names.

Reply:
Remember when they changed their name?

Reply:
Nope I have swiss cheese memory. 

*AL! ALLL!!! Does ziggy say I can leap yet, or do I need to keep helping this project?*